modules/tasks/tasks/src/main/resources/webapp/js/controllers.js

```diff
@@ -1332,27 +1332,48 @@
     });
 
-    controllers.controller('TriggersModalCtrl', function ($scope, $rootScope, $timeout, LoadingModal, BootstrapDialogManager, Triggers, ModalFactory) {
+    controllers.controller('TriggersModalCtrl', function ($scope, $rootScope, $timeout, LoadingModal, BootstrapDialogManager, Triggers, ModalFactory, $filter) {
 
-        $scope.reloadLists = function(staticTriggersPage, dynamicTriggersPage) {
+        $scope.searchTrigger = {
+            displayName: ""
+        };
+
+        $scope.changeCurrentPage = function(staticTriggersPage, dynamicTriggersPage) {
             if ($scope.validatePages(staticTriggersPage, dynamicTriggersPage)) {
-                LoadingModal.open();
-                Triggers.get(
-                {
-                    moduleName: $scope.selectedChannel.moduleName,
-                    staticTriggersPage: staticTriggersPage,
-                    dynamicTriggersPage: dynamicTriggersPage
-                },
-                function(data) {
-                    $scope.dynamicTriggers = data.dynamicTriggersList;
-                    $scope.staticTriggers = data.staticTriggersList;
-                    $scope.staticTriggersPage = $scope.staticTriggers.page;
-                    $scope.dynamicTriggersPage = $scope.dynamicTriggers.page;
-                    LoadingModal.close();
-                });
+                $scope.staticTriggers.page = staticTriggersPage;
+                $scope.dynamicTriggers.page = dynamicTriggersPage;
             }
         };
 
+        $scope.$watch('searchTrigger.displayName', function() {
+            var searchedTriggers = [];
+
+            if ($scope.hasStaticTriggers) {
+                $scope.staticTriggers.page = 1;
+                if ($scope.searchTrigger.displayName !== "") {
+                    searchedTriggers = $filter('filter')($scope.staticTriggers.triggers, $scope.searchTrigger);
+                    $scope.staticTriggers.total = parseInt(searchedTriggers.length / $scope.pageSize, 10);
+                    if (searchedTriggers.length % $scope.pageSize > 0) {
+                        $scope.staticTriggers.total += 1;
+                    }
+                } else {
+                    $scope.staticTriggers.total = $scope.staticTriggersPages;
+                }
+            }
+            if ($scope.hasDynamicTriggers) {
+                $scope.dynamicTriggers.page = 1;
+                if ($scope.searchTrigger.displayName !== "") {
+                    searchedTriggers = $filter('filter')($scope.dynamicTriggers.triggers, $scope.searchTrigger);
+                    $scope.dynamicTriggers.total = parseInt(searchedTriggers.length / $scope.pageSize, 10);
+                    if (searchedTriggers.length % $scope.pageSize > 0) {
+                        $scope.dynamicTriggers.total += 1;
+                    }
+                } else {
+                    $scope.dynamicTriggers.total = $scope.dynamicTriggersPages;
+                }
+            }
+        });
```

Comment:
this two if conditions look almost the same, the only change is dynamic/static. I think it can be done simplier